TilerServer/TilerWebModule.pas

```diff
@@ -3646,39 +3674,16 @@ function TSliceDiffGeometryICLR.GenerateTileCalc(const aExtent: TExtent; aBitmap
             else
             begin
               //see if we can find L width
-              if not IsNaN(isgop.Value.value) then
-              begin
-                if not IsNaN(refObj.value)
-                then widthL := (isgop.Value.value + refObj.value) / 2
-                else widthL := isgop.Value.value;
-              end
-              else
-              begin
-                if not IsNaN(refObj.value)
-                then widthL := refObj.value
-                else validL := False;
-              end;
+              widthL := CalculateWidth(isgop.Value.value, refObj.value, validL);
+
               //see if we can find R width
-              if not IsNaN(isgop.Value.value2) then
-              begin
-                if not IsNaN(refObj.value2)
-                then widthR := (isgop.Value.value2 + refObj.value2) / 2
-                else widthR := isgop.Value.value2;
-              end
-              else
-              begin
-                if not IsNaN(refObj.value2)
-                then widthR := refObj.value2
-                else validR := False;
-              end;
+              widthR := CalculateWidth(isgop.Value.value2, refObj.value2, validR);
+
               //see if we can find L color
-              if not (IsNaN(isgop.Value.texture) or IsNaN(refObj.texture))
-              then colorsL := fPalette.ValueToColors(isgop.Value.texture-refObj.texture)
-              else validL := False;
+              colorsL := GetPaletteColor(isgop.Value.texture, refObj.texture, validL);
+
               //see if we can find R color
-              if not (IsNaN(isgop.Value.texture2) or IsNaN(refObj.texture2))
-              then colorsR := fPalette.ValueToColors(isgop.Value.texture2-refObj.texture2)
-              else validR := False;
+              colorsR := GetPaletteColor(isgop.Value.texture2, refObj.texture2, validR);

```

Comment:
'isgop' is not a very good name for ActiveScenario!

Reply:
Fixed!